core/server/src/test/java/alluxio/master/file/PermissionCheckerTest.java

```diff
@@ -194,6 +194,14 @@ public void before() throws Exception {
     mPermissionChecker = new PermissionChecker(sTree);
   }
 
+  // Helper function to create a path and set the permission to what specified in option.
+  private static void createAndSetPermission(String path, CreateFileOptions option)
+      throws Exception {
+    InodeTree.CreatePathResult result = sTree.createPath(new AlluxioURI(path), option);
+    result.getCreated().get(result.getCreated().size() - 1)
```

Comment:
Yes, I think you get the point --- basically user/group and permission bits are filled/used in different code path. I am cool with you to take on the improvement in followup PRs, and happy to help out.